File: p3/4.cpp
#include <iostream>
using namespace std;

int factorial(int n);
int long comb(int n, int x);

int main(int argc, char *argv[]) {
  int n, x;
  cin >> n;
  cin >> x;

  int long r = comb(n, x);
  cout << r << "\n";

  return 0;
}

int long comb(int n, int x) {
  int long nf = factorial(n);
  int long xf = factorial(x);
  int n_xf = factorial(n - x);
  // cout << "nf: " << nf << "\n";
  // cout << "xf: " << xf << "\n";
  // cout << "n_xf: " << n_xf << "\n";
  return nf / (xf * n_xf);
}

int factorial(int n) {
  int buff = 1;

  for (int i = 1; i <= n; i++) {
    buff *= i;
  }
  return buff;
}

File: p3/4.py
def factorial(n):
    buff = 1
    for i in range(1, n + 1):
        buff *= i

    return buff


def comb(n, x):
    nf = factorial(n)
    xf = factorial(x)
    n_xf = factorial(n - x)

    return nf / (xf * n_xf)


def main():
    n = int(input())
    x = int(input())

    print(int(comb(n, x)))


if __name__ == "__main__":
    main()
